x/programs/cmd/simulator/src/lib.rs

```diff
@@ -207,19 +243,26 @@ pub enum ClientError {
     Read(#[from] std::io::Error),
     #[error("Deserialization error: {0}")]
     Deserialization(#[from] serde_json::Error),
+    #[error("Borsh deserialization error")]
+    BorshDeserialization,
```

Comment:
`borsh::io::Error` is a subset of `std::io::Error` so I couldn't add the `#[from]` variant or it would overlap with the `Read` error.